### PR TITLE
Refactor msr core

### LINKED ISSUE #239: Refactor msr_safe.[ch] into single directory
The `msr_safe.[ch]` files were copied into the `AMD` directory to get that driver up and running.  Propose moving these files out of the `Intel` and `AMD` directories and into their own `msr` directory to avoid drift (and make writing the hardware emulator a little easier).

### PR DESCRIPTION
**Pull request by rountree**
Fixes #239

### CODE REVIEW

File: src/variorum/msr/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  CACHE INTERNAL "")
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${variorum_includes})
+
+#add_library(variorum_intel OBJECT
+#            ${variorum_intel_sources}
+#            ${variorum_intel_headers})
+
+### Shared libraries need PIC
+#set_property(TARGET ${variorum_intel} PROPERTY POSITION_INDEPENDENT_CODE 1)
```
Comment (slabasan) [inline src/variorum/msr/CMakeLists.txt]: Comments in here can be removed.

Comment (rountree) [reply src/variorum/msr/CMakeLists.txt:21]: @slabasan Want me to kill off the commented-out code?

Comment (rountree) [reply src/variorum/msr/CMakeLists.txt]: Done, pushed.

Comment (rountree) [reply src/variorum/msr/CMakeLists.txt]: @slabasan I'm out of practice.  I didn't want the Lupine commit to go in.  Let me try a rebase and consolidate.

**Pull request merged**